packages/foundry/test/ReBalancerE2E.t.sol

```diff
@@ -0,0 +1,285 @@
+            permit2.approve(address(bpt), address(batchRouter), type(uint160).max, type(uint48).max);
+            permit2.approve(address(bpt), address(rebalancer), type(uint160).max, type(uint48).max);
+
+            vm.stopPrank();
+        }
+    }
+
+    // Overrides pool creation to set liquidityManagement (disables unbalanced liquidity).
+    function createPool() internal override returns (address) {
+        IERC20[] memory sortedTokens = InputHelpers.sortTokens(
+            [address(dai), address(usdc)].toMemoryArray().asIERC20()
+        );
+        for (uint256 i = 0; i < sortedTokens.length; i++) {
+            poolTokens.push(sortedTokens[i]);
+            tokenAmounts.push(TOKEN_AMOUNT);
+        }
+
+        weights = [uint256(50e16), uint256(50e16)].toMemoryArray();
+
+        PoolRoleAccounts memory roleAccounts;
+        // Allow pools created by `factory` to use poolHooksMock hooks
+        // PoolHooksMock(poolHooksContract).allowFactory(address(_factoryMock));
+
+        WeightedPool newPool = WeightedPool(
+            WeightedPoolFactory(address(_factoryMock)).create(
+                "ERC20 Pool",
+                "ERC20POOL",
+                vault.buildTokenConfig(sortedTokens),
+                weights,
+                roleAccounts,
+                DEFAULT_SWAP_FEE,
+                poolHooksContract,
+                false, // Do not enable donations
+                false, // Do not disable unbalanced add/remove liquidity
+                ZERO_BYTES32
+            )
+        );
+        return address(newPool);
+    }
+
+    function testAddLiquidity() public {
+        BaseVaultTest.Balances memory balancesBefore = getBalances(bob);
+        uint256[] memory maxAmountsIn = [dai.balanceOf(bob), usdc.balanceOf(bob)].toMemoryArray();
+        vm.prank(bob);
+        uint256[] memory amountsIn = rebalancer.addLiquidityProportional(
+            pool,
+            maxAmountsIn,
+            bptAmount,
+            false,
+            bytes("")
+        );
+        vm.stopPrank();
+
+        BaseVaultTest.Balances memory balancesAfter = getBalances(bob);
+
+        // bob sends correct lp tokens
+        assertEq(
+            balancesBefore.bobTokens[daiIdx] - balancesAfter.bobTokens[daiIdx],
+            amountsIn[daiIdx],
+            "bob's DAI amount is wrong"
+        );
+        assertEq(
+            balancesBefore.bobTokens[usdcIdx] - balancesAfter.bobTokens[usdcIdx],
+            amountsIn[usdcIdx],
+            "bob's USDC amount is wrong"
+        );
+    }
+
+    function setUpRebalanceData(address pool, uint256[] memory minRatios) internal {
+        ReBalancerHook.RebalanceData[] memory data = new ReBalancerHook.RebalanceData[](2);
+        for (uint256 i = 0; i < 2; i++) {
+            data[i] = ReBalancerHook.RebalanceData({ minRatio: minRatios[i], rebalanceRequired: false });
+        }
+
+        vm.prank(address(0)); // Only pool creator can set rebalance data
```

Comment:
Interesting trick :) Not sure what you would do in production, though. Maybe you would need a custom Weighted Pool Factory that deploys the regular WeightedPool, but keeps track of the deployer in the factory, so you could query not only whether a given pool came from that factory, but who the deployer was. (In practice, you could just check that the deployer is non-zero.)

Reply:
Okay so when I was trying to find the interfaces for finding owner from the pool and only looked at code for weighted pool directly from the code of monorepo and did not have look at the factory code. I will definetly fix this